pkg/resource/account/sdk.go

```diff
@@ -0,0 +1,377 @@
+	} else {
+		ko.Status.AccountID = nil
+	}
+	if resp.CreateAccountStatus.AccountName != nil {
+		ko.Spec.AccountName = resp.CreateAccountStatus.AccountName
+	} else {
+		ko.Spec.AccountName = nil
+	}
+	if resp.CreateAccountStatus.CompletedTimestamp != nil {
+		ko.Status.CompletedTimestamp = &metav1.Time{*resp.CreateAccountStatus.CompletedTimestamp}
+	} else {
+		ko.Status.CompletedTimestamp = nil
+	}
+	if resp.CreateAccountStatus.FailureReason != nil {
+		ko.Status.FailureReason = resp.CreateAccountStatus.FailureReason
+	} else {
+		ko.Status.FailureReason = nil
+	}
+	if resp.CreateAccountStatus.GovCloudAccountId != nil {
+		ko.Status.GovCloudAccountID = resp.CreateAccountStatus.GovCloudAccountId
+	} else {
+		ko.Status.GovCloudAccountID = nil
+	}
+	if resp.CreateAccountStatus.Id != nil {
+		ko.Status.ID = resp.CreateAccountStatus.Id
+	} else {
+		ko.Status.ID = nil
+	}
+	if resp.CreateAccountStatus.RequestedTimestamp != nil {
+		ko.Status.RequestedTimestamp = &metav1.Time{*resp.CreateAccountStatus.RequestedTimestamp}
+	} else {
+		ko.Status.RequestedTimestamp = nil
+	}
+	if resp.CreateAccountStatus.State != nil {
+		ko.Status.State = resp.CreateAccountStatus.State
+	} else {
+		ko.Status.State = nil
+	}
+
+	rm.setStatusDefaults(ko)
+	return &resource{ko}, nil
+}
+
+// newCreateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Create API call for the resource
+func (rm *resourceManager) newCreateRequestPayload(
+	ctx context.Context,
+	r *resource,
+) (*svcsdk.CreateAccountInput, error) {
+	res := &svcsdk.CreateAccountInput{}
+
+	if r.ko.Spec.AccountName != nil {
+		res.SetAccountName(*r.ko.Spec.AccountName)
+	}
+	if r.ko.Spec.Email != nil {
+		res.SetEmail(*r.ko.Spec.Email)
+	}
+	if r.ko.Spec.IAMUserAccessToBilling != nil {
+		res.SetIamUserAccessToBilling(*r.ko.Spec.IAMUserAccessToBilling)
+	}
+	if r.ko.Spec.RoleName != nil {
+		res.SetRoleName(*r.ko.Spec.RoleName)
+	}
+	if r.ko.Spec.Tags != nil {
+		f4 := []*svcsdk.Tag{}
+		for _, f4iter := range r.ko.Spec.Tags {
+			f4elem := &svcsdk.Tag{}
+			if f4iter.Key != nil {
+				f4elem.SetKey(*f4iter.Key)
+			}
+			if f4iter.Value != nil {
+				f4elem.SetValue(*f4iter.Value)
+			}
+			f4 = append(f4, f4elem)
+		}
+		res.SetTags(f4)
+	}
+
+	return res, nil
+}
+
+// sdkUpdate patches the supplied resource in the backend AWS service API and
+// returns a new resource with updated fields.
+func (rm *resourceManager) sdkUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	return nil, ackerr.NewTerminalError(ackerr.NotImplemented)
+}
+
+// sdkDelete deletes the supplied resource in the backend AWS service API
+func (rm *resourceManager) sdkDelete(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkDelete")
+	defer func() {
+		exit(err)
+	}()
+	// TODO(jaypipes): Figure this out...
+	return nil, nil
+
+}
```

Comment:
Do we want to call `CloseAccount` on deletion? https://docs.aws.amazon.com/organizations/latest/APIReference/API_CloseAccount.html

Reply:
You can instruct the code generator to do so by adding something similar to https://github.com/aws-controllers-k8s/lambda-controller/blob/main/generator.yaml#L28-L33